### PR TITLE
[RFC] man.vim: default mapping and general improvements

### PR DESCRIPTION
**Pull request by nhooyr**
See https://github.com/neovim/neovim/pull/4449#issuecomment-237809542

All I've done thus far is make `<Plug>(Man)` use `<cword>` for non-man buffers (and some small improvements to other parts of the code), just like `keywordprg` because the syntax is unpredictable. But in man buffers, we use `<cWORD>`. I'm not sure how to continue because I am very unfamiliar with c and it looks like the default `K` mapping is hard coded in c.

Also, was man.vim the only reason `keywordprg` accepted commands? If so, I think we should remove that behaviour.

### CODE REVIEW

**Comment by nhooyr** [comment]
Actually, I'm not so sure about using `<cword>` for other filetypes.
Say there is a comment in a C file with the manpage refrence:`printf(3)`. If you press `K` on it, it will actually take you to `printf` because `expand('<cword>')` returned `printf`. And if you're on the 3, it will take you nowhere and print a not found error.

**Comment by justinmk** [comment]
> was man.vim the only reason keywordprg accepted commands? If so, I think we should remove that behaviour.

Why?

**Comment by nhooyr** [comment]
Alright, forget all that, I'm very sleepy :(. I've made the command work with `keywordprg`. Turns out, `v:count1` is available for commands as well so I can use the same trick I used with the mappings to ensure the user explicitly set the section.

When you open a man buffer though, the mapping for `K` switches to the one that uses `<cWORD>`. This works better for manpages because I'd rather not try and compile a list of allowable characters in manpage names.

This way, the current defaults work perfectly fine with minimal change.

File: runtime/plugin/man.vim [hunk 1]
```diff
@@ -5,6 +5,12 @@ if exists('g:loaded_man')
...
 
-command! -complete=customlist,man#complete -nargs=* Man call man#open_page_command(<f-args>)
+command! -count=0 -complete=customlist,man#complete -nargs=* Man call man#open_page_command(v:count, v:count1, <f-args>)
 
-nnoremap <silent> <Plug>(Man) :<C-U>call man#open_page_mapping(v:count, v:count1, expand('<cWORD>'))<CR>
+autocmd BufEnter man://*
+      \ if exists('b:manwidth') && line('$') == 1 && empty(getline(1)) |
+      \   unlet b:manwidth |
+      \   silent execute 'Man' matchstr(expand('<amatch>'), 'man://\zs.*') |
+      \ endif
```
Comment (justinmk) [inline runtime/plugin/man.vim]: This seems like a hack. Why is the buffer empty?

Comment (nhooyr) [inline runtime/plugin/man.vim]: I'm gonna actually leave this out for another PR.

Comment (tweekmonster) [inline runtime/plugin/man.vim]: @justinmk It was a suggestion I made in #5169.  Out of habit, I use `:e` to reload a buffer.  For me, it made sense to do that when I resized the window and wanted the man page to be re-rendered.

**Comment by nhooyr** [comment]
For posterity, https://github.com/nhooyr/neovim/commit/f34319df032ba35075b4410287c99df0ee29d651#commitcomment-18549463

**Comment by justinmk** [comment]
re https://github.com/nhooyr/neovim/commit/f34319df032ba35075b4410287c99df0ee29d651#commitcomment-18549824

- Help files are not restored in sessions, unless a flag is set. 
- It makes sense to list manpage buffers for  `:ls`, `:bnext`, etc.
- With `buftype=` , what happens when you do `:confirm qa` ? Does it confirm closing the manpage? 

I think not supporting sessions is the right thing to do for now. Let's try to fix the general problem of why sessions aren't working with `buftype=` instead of making man.vim have incorrect settings. It may requires changing the session restore logic, even. (Separate PR, of course)

**Comment by nhooyr** [comment]
No it does not confirm closing the manpage.

Alright, now `buftype=nofile` is set and the buffers are listed.
